theme/_router.ts

```diff
@@ -0,0 +1,19 @@
+import { match, route } from '@/theme-kit';
+import { assertServerEnv } from '@/utils';
+
+export default function router(path: string | string[]) {
+    assertServerEnv('router');
+
+    return match(path, [
+        route('/', () => import('./homepage')),
+        route('/[locale]', () => import('./homepage')),
+        route('/[slug]', () => import('./story')),
+        route('/s/[uuid]', () => import('./story')),
+        route('/media', () => import('./gallery')),
+        route('/[locale]/media', () => import('./gallery')),
+        route('/media/album/[uuid]', () => import('./gallery-album')),
+        route('/[locale]/media/album/[uuid]', () => import('./gallery-album')),
+        route('/search', () => import('./search')),
+        route('/[locale]/search', () => import('./search')),
+    ]);
```

Comment:
This may become abstracted one level further: by hiding the actual routes behind the default setup (which is unlikely to change from theme to theme).